mm2src/mm2_tests.rs

```diff
@@ -8282,6 +8282,48 @@ fn test_mm2_db_migration() {
     .unwrap();
 }
 
+#[test]
+#[cfg(not(target_arch = "wasm32"))]
+fn test_get_public_key() {
+    let coins = json!([
+        {"coin":"RICK","asset":"RICK","rpcport":8923,"txversion":4,"protocol":{"type":"UTXO"}},
+    ]);
+
+    let mm = MarketMakerIt::start(
+        json!({
+            "gui": "nogui",
+            "netid": 9998,
+            "passphrase": "bob passphrase",
+            "rpc_password": "password",
+            "coins": coins,
+            "i_am_seed": true,
+        }),
+        "password".into(),
+        None,
+    )
+    .unwrap();
+    let (_dump_log, _dump_dashboard) = mm.mm_dump();
+    log!({"Log path: {}", mm.log_path.display()});
+    fn get_public_key_bot_rpc(mm: &MarketMakerIt) -> (StatusCode, String, HeaderMap) {
+        block_on(mm.rpc(json!({
+                 "userpass": "password",
+                 "mmrpc": "2.0",
+                 "method": "get_public_key",
+                 "params": {},
+                 "id": 0})))
+        .unwrap()
+    }
+    let resp = get_public_key_bot_rpc(&mm);
+
+    // Must be 200
+    assert_eq!(resp.0, 200);
+    let v: Json = serde_json::from_str(&*resp.1).unwrap();
```

Comment:
Please add a separate struct to `mm2_tests::structs` for the response similar to already existing. It also serves a purpose to avoid backward compatibility breakage.